data/content_scripts/posting_process/context_menu.js

```diff
@@ -35,7 +35,7 @@ postingProcess.menuClickHandler = function(node, data) {
     text: selectionText,
     pageURL: window.location.href,
   };
-  self.postMessage(info);
+  self.postMessage({name: "startPostingProcess", content: info});
```

Comment:
> Regarding the second point, the listener function runs in the background context?

Yeah. But the details of receiving messages from the context menu content script is hidden. For web page content script instances(workers) we can write `worker.port.on("PRIVLY_MESSAGE", handler)` to receive the messages but with context menus, we can only pass the handler as an "onMessage" option in the menu initialization.